### PR TITLE
[PR 1] DEV-169 improve BLCU download logic

### PR DESCRIPTION
**Pull request by jmaralo**
*Accidentally I also had some changes from DEV-167, any file outside the `blcu` folder is safe to ignore*

The previous BLCU download implementation just sent the data received to the frontend so it could download it to the client computer. Since that logic is not actually implemented in the frontend and it might not be desirable to implement, currently that data is stored locally at the backend, inside a donwloads folder with the contents of all the downloaded code.

In case the file creation returns an error, the frontend is notified with an error altough the file was correctly fetched from the board.

### CODE REVIEW

File: src/blcu/blcu.go [hunk 1]
```diff
@@ -26,15 +27,35 @@ type BLCU struct {
...
 }
 
+func getBoardToId(boardToIdStr map[string]string) map[string]uint16 {
+	boardToId := make(map[string]uint16)
+
+	for name, idStr := range boardToIdStr {
+		id, err := strconv.Atoi(idStr)
+
+		if err != nil {
+			//TODO: TRACE
```
Comment (smorsua) [inline src/blcu/blcu.go:49]: remove todo

File: src/blcu/download.go [hunk 2]
```diff
@@ -15,26 +17,26 @@ type downloadRequest struct {
 	Board string `json:"board"`
 }
 
-func (blcu *BLCU) handleDownload(payload json.RawMessage) ([]byte, error) {
+func (blcu *BLCU) handleDownload(payload json.RawMessage) (string, []byte, error) {
```
Comment (smorsua) [inline src/blcu/download.go:20]: Create struct DownloadResponse:

`type DownloadResponse struct {
    board string
    bin []byte
}`

File: src/blcu/upload.go [hunk 3]
```diff
@@ -13,24 +14,31 @@ import (
...
+	if err := json.Unmarshal(payload, &request); err != nil {
 		blcu.trace.Error().Err(err).Stack().Msg("Unmarshal payload")
 		return err
 	}
 
-	if err := blcu.requestUpload(uploadData.Board); err != nil {
+	id, ok := blcu.boardToId[request.Board]
+
+	if !ok {
+		//TODO: trace
```
Comment (smorsua) [inline src/blcu/upload.go:32]: remove comment

File: src/common/fs.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package common
```
Comment (smorsua) [inline src/common/fs.go:1]: Delete this file if the function is not used anywhere

**Review by smorsua — COMMENTED** [review]
Some small suggestions

**Review by smorsua — APPROVED** [review]

**Review by lipezaballa — APPROVED** [review]

**Pull request merged**